src/extension/extension.ts

```diff
@@ -567,9 +564,7 @@ export async function activate(ctx: sourcegraph.ExtensionContext): Promise<void>
                         logger.log('Definition trace', span.generateTraceURL())
                     }
 
-                    const lsifResult = (await isLSIFAvailable(textDocument))
-                        ? await lsif.definition(textDocument, position)
-                        : undefined
+                    const lsifResult = await lsif.definition(textDocument, position)
```

Comment:
Just to confirm, in the case LSIF data doesn't exist, this returns `undefined` so that it can fallback to the language server?

Reply:
Yes.

Reply:
The lsif providers will query the GraphQL API if it's a version that's new enough to have it, otherwise it will call the isLSIFAvailable function and use the old HTTP API. If the GraphQL API has no LSIF data for that file, it returns undefined. If isLSIFAvailable return false, it returns undefined.